File: adventofcode-2024/src/test/java/nl/mvdr/adventofcode/adventofcode2024/day15/Part2Test.java
package nl.mvdr.adventofcode.adventofcode2024.day15;

import java.util.stream.Stream;

import org.junit.jupiter.params.provider.Arguments;

import nl.mvdr.adventofcode.solver.SolverTest;

public class Part2Test extends SolverTest<Part2> {

    public Part2Test() {
        super(Part2.class);
    }
    
    static Stream<Arguments> testSolution() {
        return Stream.of(
                Arguments.of("9021", "example-day15-2024-0.txt"),
                Arguments.of("618", "example-day15-2024-2.txt"));
    }
}

File: adventofcode-2025/src/test/kotlin/nl/mvdr/adventofcode/adventofcode2025/day01/Part2Test.kt
package nl.mvdr.adventofcode.adventofcode2025.day01

import nl.mvdr.adventofcode.solver.FunctionSolverTest
import org.junit.jupiter.params.provider.Arguments

class Part2Test: FunctionSolverTest<Int>(::solvePart2) {
    companion object {
        @JvmStatic
        fun testSolution(): List<Arguments> {
            return listOf(
                Arguments.of("6", "example-day01.txt")
            )
        }
    }
}